Mathlib/Data/Bitvec/Lemmas.lean

```diff
@@ -124,4 +124,44 @@ theorem ofFin_toFin {n} (v : BitVec n) : ofFin (toFin v) = v := by
   rfl
 #align bitvec.of_fin_to_fin Std.BitVec.ofFin_toFin
 
+/-!
+  ## Distributivity of ofFin
+  We add simp-lemmas that show how `ofFin` distributes over various bitvector operations, showing
+  that bitvector operations are equivalent to `Fin` operations
+-/
+@[simp] lemma neg_ofFin (x : Fin (2^w)) : -(ofFin x) = ofFin (-x) := by
+  rw [neg_eq_zero_sub]; rfl
+
+@[simp] lemma ofFin_and_ofFin (x y : Fin (2^w)) : ofFin x &&& ofFin y = ofFin (x &&& y) := rfl
+@[simp] lemma ofFin_or_ofFin  (x y : Fin (2^w)) : ofFin x ||| ofFin y = ofFin (x ||| y) := rfl
+@[simp] lemma ofFin_xor_ofFin (x y : Fin (2^w)) : ofFin x ^^^ ofFin y = ofFin (x ^^^ y) := rfl
+@[simp] lemma ofFin_add_ofFin (x y : Fin (2^w)) : ofFin x + ofFin y   = ofFin (x + y)   := rfl
+@[simp] lemma ofFin_sub_ofFin (x y : Fin (2^w)) : ofFin x - ofFin y   = ofFin (x - y)   := rfl
+@[simp] lemma ofFin_mul_ofFin (x y : Fin (2^w)) : ofFin x * ofFin y   = ofFin (x * y)   := rfl
```

Comment:
(if the user wanted to work with `Fin` instead, then they could apply `toFin_injective`, and push `toFin` to the inside in the same way).

Reply:
My reason for the direction is that `ofFin x &&& ofFin y` unfolds to `(ofFin x).toFin &&& (ofFin y).toFin`, where `(ofFin x).toFin = x`. So, the `Fin` expression is (in my mind) simpler / more fundamental. Turning that aroung seems to me not something we'd want `simp` to do by default.

I'll concede that I often forgot how useful `toFin_injective` can be, but it cannot always be used. For example, for inequalities or in subexpressions. Then again, in my current use-case it can, so I'm more than happy to leave these theorems out if you insist they are not desirable

Reply:
I'd prefer that we keep the *reverse* of these theorems; they match things like `MulOpposite.op_add` etc.